pkg/pillar/zedcloud/send.go

```diff
@@ -749,7 +754,11 @@ func SendOnIntf(workContext context.Context, ctx *ZedCloudContext, destURL strin
 				return d.Dial(network, address)
 			}
 			r := net.Resolver{Dial: resolverDial, PreferGo: true, StrictErrors: false}
-			d := net.Dialer{Resolver: &r, LocalAddr: &localTCPAddr}
+			d := net.Dialer{
+				Resolver:  &r,
+				LocalAddr: &localTCPAddr,
+				Timeout:   clientConfig.TCPHandshakeTimeout,
```

Comment:
Doesn't this timeout apply to setting up the TLS connection and not just the TCP connection? The TLS handshake can take a lot longer than the SYN exchange since it needs significant user-level CPU on client and server.

Reply:
FWIW https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/ says it is just for the TCP connect and not TLS so things are good.